content-repo/extra-docs/articles/troubleshooting.md

```diff
@@ -0,0 +1,83 @@
+
+
+If the integration/automation is using http based communication we recommend testing as a first step locally via the `curl` utility if it is possible to perform network communication with the http endpoint. Run the `curl` command on the Server or Engine machine by logging in via ssh. Common `curl` command variants (`httpbin.org` is used as an example url):
+```bash
+# Run simple curl command with -v for verbose output:
+curl -v https://httpbin.org/status/200
+
+# Run with -k to trust any certificate in case you receive errors regarding certificates
+curl -vk https://httpbin.org/status/200
+
+# curl will use the machine env variables for proxy settings. If you wish to ignore the proxy settings run:
+curl -vk --noproxy "*" https://httpbin.org/status/200
+
+# Setting explicitly a proxy server to use by curl
+curl -x http://192.168.0.1:8080 https://httpbin.org/status/200
+
+# Passing an additional header as part of the curl request:
+curl -v -H 'Accept: application/json' https://httpbin.org/headers
+
+# In cases that the integration uses basic authentication, you can also easily test the credentials:
+curl -v --user myuser:mypass https://httpbin.org//basic-auth/myuser/mypass
+```
+
+More info about `curl` is available at [Everything curl](https://ec.haxx.se/).
+
+If you are not able to perform a basic `curl` request from the machine to the target http endpoint, the issue is probably not a problem with the integration/automation but rather with the networking setup of the Server/Engine machine. Make sure to first resolve the networking issue so a basic `curl` command succeeds before continuing to test the integration/automation. Many times this resolves to a firewall, NAT or proxy issue. 
+
+### Docker Based Networking
+Docker Based integrations/automations are written in Python or Powershell. They can be identified by inspecting the integration/automation settings and under *Language type* will appear **Python/Powershell**. Docker creates its own networking, thus the integrations/automations are using a different networking stack from the Cortex XSOAR Server/Engine. The source IPs for these integrations/automations are different and provided according to the Docker networking configuration.
+
+As with [Host Based Networking](#host-based-networking), for integrations/automations that use http endpoints we recommend testing as a first step with `curl` from within a docker container. This can be done by logging in to the Server/Engine machine via ssh and running the following command:
```

Comment:
```suggestion
As with [Host Based Networking](#host-based-networking), for integrations/automations that use HTTP endpoints we recommend testing as a first step with `curl` from within a docker container. This can be done by logging in to the Server/Engine machine via SSH and running the following command:
```